static/js/containers/DashboardPage.js

```diff
@@ -382,6 +411,17 @@ class DashboardPage extends React.Component {
     this.context.router.push("/learner");
   };
 
+  shouldSkipFinancialAid = (): boolean => {
+    return R.compose(
+      R.any(coupon => (
+        coupon.content_type === COUPON_CONTENT_TYPE_PROGRAM &&
```

Comment:
I think you also need to make sure that `coupon.program_id` equals the currently selected program id